docs/user-guide.md

```diff
@@ -15,37 +15,35 @@ Such annotations are useful to (for example):
 
 ## Java variables are references
 
-In Java, all variables, except primitives like `int`, are references. We often
-think of `String x;` as meaning "`x` is a `String`", but actually it means "`x` is a
-_reference_ to a `String`". One important difference between these two views is that
-`x` can also be null. The goal of JSpecify is to move towards a world where `String
-x;` means "`x` is a reference to an actual string", and if you want a reference
-that might be null then you write `@Nullable String x;`.
+In Java, all non-primitive variables are references. We often think of `String
+x;` as meaning "`x` is a `String`", but actually it means "`x` is a _reference_
+to a string or it is null". JSpecify includes a notion of "null-marked code"
+where `String x;` means "`x` is a reference to an actual string", and if you
```

Comment:
To emphasize the distinction, consider changing ", and" to a simple semicolon.

Or maybe refactor to something like:

JSpecify includes a `@NullMarked` annotation. In code covered by that annotation, `String x` means "`x` is a reference to an actual string", and `@Nullable String x` means "`x` is either null or a reference to an actual string."

Reply:
OK, I think that works well. I've reworded the earlier text to be more consistent with it.